src/db/sqldb/metrics_sqldb.go

```diff
@@ -0,0 +1,130 @@
+package sqldb
+
+import (
+	"database/sql"
+	"encoding/json"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+	_ "github.com/lib/pq"
+
+	"db"
+	"metricscollector/metrics"
+)
+
+type MetricsSQLDB struct {
+	sqldb  *sql.DB
+	logger lager.Logger
+	url    string
+}
+
+func NewMetricsSQLDB(url string, logger lager.Logger) (*MetricsSQLDB, error) {
+	metricsDB := &MetricsSQLDB{
+		logger: logger,
+		url:    url,
+	}
+
+	var err error
+	metricsDB.sqldb, err = sql.Open(db.PostgresDriverName, url)
+	if err != nil {
+		logger.Error("open-metrics-db", err, lager.Data{"url": url})
+		return nil, err
+	}
+
+	err = metricsDB.sqldb.Ping()
+	if err != nil {
+		metricsDB.sqldb.Close()
+		metricsDB.sqldb = nil
```

Comment:
your are right.  We will never get a  metricsDB with a nil metricsDB.sqldb as long as it is created from NewMetricsSQLDB(), not initialized directly from struct MetricsSQLDB